src/components/Login/Login.jsx

```diff
@@ -68,10 +71,20 @@ export class Login extends React.Component {
                 username: this.state.value
             })
             .then(response => {        
-                if (response.status === 200) {                    
-                    this.setState({
-                        loggedIn: true
+                if (response.status === 200) {
+                    const socket = connectSocket();
+                    return new Promise((resolve, reject) => {
```

Comment:
Maybe I don't have the context but this looks like something that should happen in `connectSocket`. What is actually happening here.